### PR TITLE
Handle both restart and start, tested

### PR DESCRIPTION
**Pull request by jgminer**

### CODE REVIEW

File: server/zoom/agent/entities/application.py [hunk 1]
```diff
@@ -558,13 +558,38 @@ def _init_actions(self, settings):
...
 
-        if start_action is None:
-            self._read_only = True
-        elif start_action.disabled is True:
+        # Two special cases - both start and restart and neither
+        if start_action is not None and restart_action is not None:
+            if start_action.disabled is True and restart_action.disabled is True:
+                self._read_only = True
+            else:
+                self._log.warning('Start and restart have opposite settings, readonly shown as False')
```
Comment (nicmoski) [inline server/zoom/agent/entities/application.py]: I don't think this should be a warning. This is a valid configuration.

File: server/zoom/agent/entities/application.py [hunk 2]
```diff
@@ -558,13 +558,38 @@ def _init_actions(self, settings):
...
+        # Sentinel config may include either start or restart blocks, if either are disabled show as read-only
         start_action = actions.get('start', None)
+        restart_action = actions.get('restart', None)
 
-        if start_action is None:
-            self._read_only = True
-        elif start_action.disabled is True:
+        # Two special cases - both start and restart and neither
+        if start_action is not None and restart_action is not None:
+            if start_action.disabled is True and restart_action.disabled is True:
```
Comment (nicmoski) [inline server/zoom/agent/entities/application.py]: https://www.python.org/dev/peps/pep-0008/#programming-recommendations
Avoid `is True`

File: server/zoom/common/constants.py [hunk 3]
```diff
@@ -3,11 +3,7 @@
 
 __env_connections = {
     "local": "localhost:2181",
-    "Staging": ('ZooStaging01:2181,'
-                'ZooStaging02:2181,'
-                'ZooStaging03:2181,'
-                'ZooStaging04:2181,'
-                'ZooStaging05:2181'),
+    "Staging": ('zk:2181'),
```
Comment (jalons) [inline server/zoom/common/constants.py]: This will need to be backed out. It's left over from your using the unreleased dockerized version.

Comment (jgminer) [inline server/zoom/common/constants.py]: Thanks for catching, thought I had that file checked out separately.

**Comment by jgminer** [comment]
Everyone ok with this change, now?

**Comment by jalons** [comment]
Are we leaving the is True checks in?

**Comment by jgminer** [comment]
Sorry my git foo betrayed me once again. I'll update without the is Trues tomorrow.

File: server/zoom/agent/entities/application.py [hunk 4]
```diff
@@ -134,7 +134,7 @@ def app_details(self):
     def run(self):
         """
         - Start the zookeeper client
-        - Check for already running instances.
```
Comment (dshrader38) [inline server/zoom/agent/entities/application.py]: Good job removing the dangling space.

File: server/zoom/agent/entities/application.py [hunk 5]
```diff
@@ -199,7 +199,7 @@ def unregister(self, **kwargs):
     @catch_exception(RuntimeError)
     def uninitialize(self):
         """
-        Gracefully stop this Zookeeper session, then free any resentinels
```
Comment (dshrader38) [inline server/zoom/agent/entities/application.py]: Same goes for here too.

**Comment by jalons** [comment]
LGTM

**Comment by jalons** [comment]
Hope all is going well in your world @dshrader38

**Comment by dshrader38** [comment]
Everything is good @jalons. We should grab a drink soon. Have you been attending any DevOps meetups? I'm glad to see Zoom is still going strong, too bad it can't be used elsewhere - I have firms that would be interested in using it.

**Comment by jalons** [comment]
Funny you bring that up, as we've been talking about how we can decouple the Spot-isms from this and turn it loose, or if we should bring it back in house. I haven't made a lot of the devops meetups, but am looking forward to the new devops bookclub (starting up )[https://www.meetup.com/devops/events/233434069/]. Also need to get another SaltStack Chicago meetup planned. Always down to grab a drink, too.

**Pull request merged**